src/Hamiltonians/ExtendedHubbardReal1D.jl

```diff
@@ -15,18 +15,30 @@ Implements the extended Hubbard model on a one-dimensional chain in real space.
 * `t`: the hopping strength
 
 """
-struct ExtendedHubbardReal1D{TT,A<:SingleComponentFockAddress,U,V,T} <: AbstractHamiltonian{TT}
+struct ExtendedHubbardReal1D{TT,A<:SingleComponentFockAddress,U,V,T,PITWISTED,HARDWALL} <: AbstractHamiltonian{TT}
     add::A
 end
 
 # addr for compatibility.
-function ExtendedHubbardReal1D(addr; u=1.0, v=1.0, t=1.0)
-    U, V, T = promote(float(u), float(v), float(t))
-    return ExtendedHubbardReal1D{typeof(U),typeof(addr),U,V,T}(addr)
+function ExtendedHubbardReal1D(addr; u=1.0, v=1.0, t=1.0, pitwisted::Bool = false, hardwall::Bool = false)
```

Comment:
I think it would be nicer to have only one argument for the boundary condition:
```suggestion
function ExtendedHubbardReal1D(addr; u=1.0, v=1.0, t=1.0, boundary_condition = :periodic)
```
You can make this accept `:hard_wall` or `:twisted` as an alternative answer. In fact we might (later) want to allow passing an arbitrary phase. (This would make the Hamiltonian complex valued, so I don't recommend doing it now, even though it will be a useful feature to have.)

Please also modify the docstring such that it is consistent with the available keyword arguments.